basic-javascript/project-1.js

```diff
@@ -31,22 +58,44 @@ const greaterThanFifty = num => {
 };
 
 const add = (x, y) => {
-  return x + y;
+  if (typeof(x) !== 'number' || typeof(y) !== 'number'){
+    return 'values must both be numbers';
+  }
+  const a = x || 0;
+  const b = y || 0;
+  return a + b;
 };
 
 const subtract = (x, y) => {
-  return x - y;
+  if (typeof(x) !== 'number' || typeof(y) !== 'number'){
+    return 'values must both be numbers';
+  }
+  const a = x || 0;
+  const b = y || 0;
+  return a - b;
 };
 
 const divide = (x, y) => {
+  if ((!x && x !==0) || (!y && y!==0)) return 'must have two numbers'
```

Comment:
I am wondering - is it necessary to say both if !x && x !==0? If there is no x then it can't be zero, I think?? Or maybe I am overthinking this.

Reply:
Yes.  I played around with a few ways to do this.  I don't think it is necessary.  I was able to achieve the same functionality without the !==0.